routes/users.js

```diff
@@ -20,53 +22,42 @@ router.get("/users", async (req, res) => {
 
 // Create a new user
 router.post("/users", async (req, res) => {
-    const { name, email, about, languages} = req.body;
+    const { name, email, about, languages } = req.body;
     console.log(req.body); // Log the entire request body
     try {
         // todo: firebase will only know user's email, we will need to get user's application by email
         // and populate user entity with that data here 
-        const result = await pool.query(
-            "INSERT INTO users (name, email, about, languages) VALUES ($1, $2, $3, $4) RETURNING id",
-            [name, email, about, languages]
-        );
-
-        let userId = result.rows[0].id;
-        
+        const userId = await userService.createNewUser(name, email, about, languages);
         // todo add transactions: if something went wrong here, the user should not be saved
-        await userService.assignRoleToUser(userId, 'member');
 
-        res.status(201).json({ id: userId});
+        //muffin: i'm dying, while i was debugging assignRoleToUser it would go to the catch block
+        //and with no transactions the server instance would freeze and now i have like 30+ users with no roles in the db xD 
+        //send help, the users are orphans now
+        await userService.assignRoleToUser(userId, 'member');
+        res.status(201).json({ id: userId });
     } catch (err) {
-        console.error(err)
+        console.error(err);
         if (isUniqueConstraintViolation(err.code)) {
-            return respondWithError(res, 409, "User with this email is already registered")
+            return respondWithError(res, 409, "User with this email is already registered");
         } else if (isNotNullConstraintViolation(err.code)) {
-            return respondWithError(res, 400, err.message)
+            return respondWithError(res, 400, err.message);
         }
         respondWithError(res);
     }
 });
 
+
 // Get user by ID
 router.get("/users/:id", async (req, res) => {
     const { id } = req.params;
-
     if (!isValidIntegerId(id)) {
         return respondWithError(res, 400, "Invalid user id supplied");
     }
-
     try {
-        const result = await pool.query("SELECT * FROM users WHERE id = $1", [id]);
-        if (result.rows.length === 0) {
+        const userInfo = await userService.getUserById(id);
+        if (!userInfo) {
```

Comment:
we need to get consistent with these checks. i have already seen
if result.lenth === 0
if result.rows.length === 0
if !result

Reply:
i thought result.length implies that we deal with an array of objects, no?
[ ] doesn't mean it's null like with ! 
and we're getting userInfo as just one object, not an array
I'm no JS expert I'm sorry 😭